webrender/src/frame.rs

```diff
@@ -328,6 +328,33 @@ impl Frame {
                 continue;
             }
 
+            let mut delta = match scroll_location {
+                ScrollLocation::Delta(delta) => delta,
+                ScrollLocation::Start => {
+                    if layer.scrolling.offset.y.round() <= 0.0 {
+                        // Nothing to do on this layer.
+                        continue;
+                    }
+
+                    layer.scrolling.offset.y = 0.0;
+                    scrolled_a_layer = true;
+                    continue;
+                },
+                ScrollLocation::End => {
+                    let end_pos = -layer.content_size.height +
+                                  layer.local_viewport_rect.size.height;
```

Comment:
Let's make this layer.local_viewport_rect.size.height - layer.content_size.height (just reads a bit better switching the order around).